airbyte-webapp/src/hooks/services/useConnector.test.tsx

```diff
@@ -1,93 +1,71 @@
-import { makeCacheProvider, makeRenderRestHook } from "@rest-hooks/test";
-import { act } from "@testing-library/react-hooks";
-
+import { act, renderHook } from "@testing-library/react-hooks";
 import useConnector from "./useConnector";
-import SourceDefinitionResource from "core/resources/SourceDefinition";
-import DestinationDefinitionResource from "core/resources/DestinationDefinition";
 
-jest.mock("hooks/services/useWorkspace", () => ({
-  useWorkspace: () => ({
-    workspace: {
-      workspaceId: "workspaceId",
-    },
+jest.mock("services/connector/SourceDefinitionService", () => ({
+  useSourceDefinitionList: () => ({
+    sourceDefinitions: [
+      {
+        sourceDefinitionId: "sid1",
+        latestDockerImageTag: "0.0.2",
+        dockerImageTag: "0.0.1",
+      },
+      {
+        sourceDefinitionId: "sid2",
+        latestDockerImageTag: "",
+        dockerImageTag: "0.0.1",
+      },
+    ],
+  }),
+  useUpdateSourceDefinition: () => ({
+    mutateAsync: jest.fn(),
   }),
 }));
 
-const renderRestHook = makeRenderRestHook(makeCacheProvider);
-const results = [
-  {
-    request: SourceDefinitionResource.listShape(),
-    params: { workspaceId: "workspaceId" },
-    result: {
-      sourceDefinitions: [
-        {
-          sourceDefinitionId: "sid1",
-          latestDockerImageTag: "0.0.2",
-          dockerImageTag: "0.0.1",
-        },
-        {
-          sourceDefinitionId: "sid2",
-          latestDockerImageTag: "",
-          dockerImageTag: "0.0.1",
-        },
-      ],
-    },
-  },
-  {
-    request: DestinationDefinitionResource.listShape(),
-    params: { workspaceId: "workspaceId" },
-    result: {
-      destinationDefinitions: [
-        {
-          destinationDefinitionId: "did1",
-          latestDockerImageTag: "0.0.2",
-          dockerImageTag: "0.0.1",
-        },
-        {
-          destinationDefinitionId: "did2",
-          latestDockerImageTag: "",
-          dockerImageTag: "0.0.1",
-        },
-      ],
-    },
-  },
-];
+jest.mock("services/connector/DestinationDefinitionService", () => ({
+  useDestinationDefinitionList: () => ({
+    destinationDefinitions: [
+      {
+        destinationDefinitionId: "sid1",
```

Comment:
Trying to understand this change (and the same a few lines below): The tests in this file are all skipped for now anyway, is there any specific reason we changed those IDs to overlap with the sourceDefinitionIds above? Or is this some artifact from trying to get those tests unskipped again?

Reply:
This test was disabled before. I tried to make it as close to working as possible, but the issue is that it is difficult to test that `useUpdateDestinationDefinition` was called, as we need to mock mutateAsync. I thought, that we need to improve DI a bit, so we could inject `DefinitionService` with mocks later and check the call there.

Right now, it is not that easy to fix this test ( or a bit more investigation is required ).